src/runtime_src/core/tools/common/XBUtilities.cpp

```diff
@@ -262,12 +262,10 @@ XBUtilities::xrt_xbutil_version_cmp(bool force)
     std::string drv_version = driver.get<std::string>("version", "<unknown>");
     if (drv_name.compare("xocl") == 0 && drv_version.compare("unknown") != 0 && xrt_version.compare(drv_version) != 0) {
       const auto & errMsg = boost::str(boost::format("Unexpected XRT Driver version (%s) was found. Expected %s, to match XRT tools.") % drv_version % xrt_version);
-      if (force) {
-        std::cout << errMsg << std::endl;
-        std::cout << "Bypassing mismatch with --force." << std::endl;
-      } else {
-        throw std::runtime_error(errMsg);
-      }
+      if (force)
+        std::cout << boost::format("%s\nBypassing mismatch with --force.") % errMsg << std::endl;
```

Comment:
Using `--force` to bypass will have secondary effects for certain commands. For instance, `xbmgmt program --force` causes all permission requests to be selected as yes. Would it be possible to use a add a new flag? If we do not want to add a new flag, is this a good trade off?

Reply:
I see! @chvamshi-xilinx, do you believe we should continue using the --force flag here?

Reply:
@dbenusov-xilinx it sounds like you believe that `--force` operation definition only applies to user conditional inputs.   It is not.   --force was mention to be an option to force the code to procedure forward across issues like this.  So I do believe that @rchane is using it correctly.

As for adding any new flags for messages like this, we should discuss this further to understand all of the use-cases.